linetools/spectra/io.py

```diff
@@ -252,7 +256,12 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, flux_tags=None,
             uwave = u.Quantity(wave, unit=u.AA)
         else:
             uwave = u.Quantity(wave)
+        # Sort
+        srt = np.argsort(uwave)
+        uwave = uwave[srt]
+        fx = fx[srt]
         if sig is not None:
+            sig = sig[srt]
             xspec1d = XSpectrum1D.from_array(uwave, u.Quantity(fx),
```

Comment:
Seems to do the job.

Suggestion: Wouldn't it be better to avoid code duplication and do the sorting only once? How about using a fixed naming convention for all the important attributes/parameters/arrays needed for defining XSpectrum1D and do the sorting and calling only once in the very latest stage?